src/main/java/com/pinterest/secor/message/MessageHeader.java

```diff
@@ -0,0 +1,42 @@
+package com.pinterest.secor.message;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class MessageHeader {
+
+	private final String key;
+	private final byte[] value;
+
+	public MessageHeader(String key, byte[] value) {
+		this.key = key;
+		this.value = value;
+	}
```

Comment:
Might help if you added a Copy Constructor from [`RecordHeader`](https://github.com/apache/kafka/blob/0.11.0/clients/src/main/java/org/apache/kafka/common/header/internals/RecordHeader.java)

Then you could have `record.headers().forEach(MesageHeader::new)` in other locations